src/NuGet.Core/NuGet.PackageManagement/Projects/PackagesConfigNuGetProject.cs

```diff
@@ -73,7 +73,7 @@ public PackagesConfigNuGetProject(string folderPath, Dictionary<string, object>
 
             PackagesConfigPath = Path.Combine(folderPath, "packages.config");
 
-            var projectName = GetMetadata<string>(NuGetProjectMetadataKeys.Name);
+            var projectName = Path.GetFileNameWithoutExtension(GetMetadata<string>(NuGetProjectMetadataKeys.Name));
```

Comment:
`GetMetadata<string>(NuGetProjectMetadataKeys.Name)` should give you project name without extension only. What case did you see returning project name with extension?

Reply:
The test that I added in NuGetUpdateCommandTests.cs exposed this case. It kept failing, saying that it couldn't find a config file for Proj1.csproj. Stepping through the nuget execution, I found that this var was being set to "proj1.csproj", and that in the test execution, Nuget.exe was trying to find "packages.proj1.csproj.config".

Reply:
Upon closer inspection, I see the code that should indeed be setting this key to the project name without the extension. Since attempting to remove the extension twice could cause problems with projects with periods in the name, I am looking for another solution.

Reply:
OK, so the class that should be setting the Name key to project-sans-path is in ProjectFactory, part of NuGet.CommandLine. However, the only place where this factory is used is in PackCommand. The UpdateCommand is getting it's copy of PackagesConfigNuGetProject from MSBuildNuGetProject, in NuGet.PackageManagement, which doesn't strip the extension from the project name. 
@jainaashish , do you have any suggestions as to how this should be resolved? I'm not sure if the UpdateCommand should use ProjectFactory, or should MSBuildNuGetProject remove the extension?

Reply:
`MSBuildNuGetProject` already takes care of removing extension, if it wasn't then that would have been a bigger issue. Were you able to repro it outside of your test as well? I feel there might be some issue with the test itself. I did try to go thorough your test case, but nothing came out explicit. So may be you can just debug your test and see how does it get `packages.proj1.csproj.config`?

Reply:
I did indeed repro outside of my test case. I created a small solution with a project that referenced an old version of Newtonsoft.Json. This project had a config file named `packages.TestProj.config`. I ran the command `nuget.exe update TestProj.sln`, and got the following error: 
`WARNING: Unable to update. The project does not contain a packages.config file.`

I then renamed the config file to `packages.TestProj.csproj.config`, and that same nuget command worked as expected, updating the version of Newtonsoft.Json that I had referenced.
These results align with what I'm seeing in my test. 

Here's what I think is going on:
The `Name` that `MSBuildNuGetProject` is using ultimately comes from`ProjectName` in  `MSBuildProjectSystem`. That class's constructor defines `ProjectName` thus:
`ProjectName = Path.GetFileName(projectFullPath);`
This should presumably be `GetFileNameWithoutExtension`. I believe the reason this hasn't shown up in testing is that `TestMSBuildNuGetProjectSystem`, the mock used in the tests, allows the test writer to tell it what the ProjectName should be. Test writers, assuming correct behavior, have always been giving it a project name with the extension removed.

Reply:
Yes that's right it should be `GetFileNameWithoutExtension` inside `MSBuildProjectSystem` and that will be the right fix.

Reply:
OK. I've implemented this fix.